config/src/main/java/org/springframework/security/config/annotation/authentication/configurers/provisioning/JdbcUserDetailsManagerConfigurer.java

```diff
@@ -191,7 +191,7 @@ public JdbcUserDetailsManager getUserDetailsService() {
 
 	protected DatabasePopulator getDatabasePopulator() {
 		ResourceDatabasePopulator dbp = new ResourceDatabasePopulator();
-		dbp.setScripts(initScripts.toArray(new Resource[initScripts.size()]));
+		dbp.setScripts(initScripts.toArray(new Resource[0]));
```

Comment:
Can you explain why you would initialize it to a 0 length array? This means that `List` must now allocate another Array of the correct size which is more garbage that would need collected.

Reply:
IntelliJ says the following:
>In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.

Other sources for this can be found here:
- https://stackoverflow.com/a/29444594/3574494
- https://shipilev.net/blog/2016/arrays-wisdom-ancients/